Source/Particles/Collision/BinaryCollision/PhotonPhoton/PhotonPhotonCollisionFunc.H

```diff
@@ -0,0 +1,244 @@
+     * @param[in] I1,I2 index arrays. They determine all elements that will be used.
+     * @param[in] soa_1,soa_2 contain the struct of array data of the two species
+     * @param[in] m1,m2 are masses.
+     * @param[in] dt is the time step length between two collision calls.
+     * @param[in] dV is the volume of the corresponding cell.
+     * @param[in] cell_start_pair is the start index of the pairs in that cell.
+     * @param[out] p_mask is a mask that will be set to true if a collision occurs for a given
+     * pair. It is only needed here to store information that will be used later on when actually
+     * creating the product particles.
+     * @param[out] p_pair_indices_1,p_pair_indices_2 arrays that store the indices of the
+     * particles of a given pair. They are only needed here to store information that will be used
+     * later on when actually creating the product particles.
+     * @param[out] p_pair_reaction_weight stores the weight of the product particles. It is only
+     * needed here to store information that will be used later on when actually creating the
+     * product particles.
+     * @param[in] engine the random engine.
+     */
+    AMREX_GPU_HOST_DEVICE AMREX_INLINE
+    void operator() (
+        index_type const I1s, index_type const I1e,
+        index_type const I2s, index_type const I2e,
+        index_type const* AMREX_RESTRICT I1,
+        index_type const* AMREX_RESTRICT I2,
+        SoaData_type soa_1, SoaData_type soa_2,
+        GetParticlePosition /*get_position_1*/, GetParticlePosition /*get_position_2*/,
+        amrex::ParticleReal const  /*q1*/, amrex::ParticleReal const  /*q2*/,
+        amrex::ParticleReal const  /*m1*/, amrex::ParticleReal const  /*m2*/,
+        amrex::Real const  dt, amrex::Real const dV,
+        index_type const cell_start_pair, index_type* AMREX_RESTRICT p_mask,
+        index_type* AMREX_RESTRICT p_pair_indices_1, index_type* AMREX_RESTRICT p_pair_indices_2,
+        amrex::ParticleReal* AMREX_RESTRICT p_pair_reaction_weight,
+        amrex::RandomEngine const& engine) const
+        {
+            amrex::ParticleReal * const AMREX_RESTRICT w1 = soa_1.m_rdata[PIdx::w];
+            amrex::ParticleReal * const AMREX_RESTRICT u1x = soa_1.m_rdata[PIdx::ux];
+            amrex::ParticleReal * const AMREX_RESTRICT u1y = soa_1.m_rdata[PIdx::uy];
+            amrex::ParticleReal * const AMREX_RESTRICT u1z = soa_1.m_rdata[PIdx::uz];
+
+            amrex::ParticleReal * const AMREX_RESTRICT w2 = soa_2.m_rdata[PIdx::w];
+            amrex::ParticleReal * const AMREX_RESTRICT u2x = soa_2.m_rdata[PIdx::ux];
+            amrex::ParticleReal * const AMREX_RESTRICT u2y = soa_2.m_rdata[PIdx::uy];
+            amrex::ParticleReal * const AMREX_RESTRICT u2z = soa_2.m_rdata[PIdx::uz];
+
+            // Number of macroparticles of each species
+            const int NI1 = I1e - I1s;
+            const int NI2 = I2e - I2s;
+            const int max_N = amrex::max(NI1,NI2);
```

Comment:
This should be slightly adapted to match the changes in:
https://github.com/BLAST-WarpX/warpx/pull/4577
In particular: 
https://github.com/BLAST-WarpX/warpx/pull/4577/files#diff-357ade83fc8e5bd11cdc99d9dacdaebfb158c8abf14ec8c1ed80965587db1b55R121
https://github.com/BLAST-WarpX/warpx/pull/4577/files#diff-4bc59234483d690ced7ddbc7e29dcbb432d0d01825eaa16d48afe47a30307438R391